security/subsystem/src/main/java/org/jboss/as/security/remoting/RemotingLoginModule.java

```diff
@@ -123,7 +123,7 @@ public boolean login() throws LoginException {
                         SSLSession session = con.getSslSession();
                         if (session != null) {
                             try {
-                                credential = session.getPeerCertificateChain()[0];
+                                credential = session.getPeerCertificates()[0];
```

Comment:
This change could not go into 6.4 anyway as it would break compatibility - a config option is the only way to retain compatibility and the default behaviour must match the old implementation.